### PR TITLE
Update to use safe scaling algorithm from Reference-LAPACK PR 527

### PR DESCRIPTION
**Pull request by martin-frbg**
fixes original topic of #4130

### CODE REVIEW

File: interface/rotg.c [hunk 1]
```diff
@@ -65,32 +85,40 @@ void CNAME(FLOAT *DA, FLOAT *DB, FLOAT *C, FLOAT *S){
...
 
-  roe = db;
-  if (ada > adb) roe = da;
 
-  if (scale == ZERO) {
+  if (adb == ZERO) {
     *C = ONE;
     *S = ZERO;
-    *DA = ZERO;
+    DA = ZERO;
```
Comment (vladimir-ch) [inline interface/rotg.c]: I think this is unintentional.

Comment (martin-frbg) [reply interface/rotg.c]: Certainly hope so...but this is just a dump of my first draft, too much work too little brain right now

File: interface/rotg.c [hunk 2]
```diff
@@ -112,7 +112,7 @@ void CNAME(FLOAT *DA, FLOAT *DB, FLOAT *C, FLOAT *S){
   if (adb == ZERO) {
     *C = ONE;
     *S = ZERO;
-    DA = ZERO;
+    *DA = ZERO;
```
Comment (vladimir-ch) [inline interface/rotg.c]: I'm not sure about this change. If the input vector is [a,0], then the rotation is the identity and the rotated vector [a,0]. Maybe your initial intention was to delete the whole line as in the other `#if` branch?

Comment (martin-frbg) [reply interface/rotg.c]: yeah, looks like a cut-n-paste gone wrong. thanks for your vigilance - still haven't run any tests with this

**Pull request merged**